packages/extension-api-caller/src/lib/filters/restApiCaller.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  FunctionalFilter,
+  InternalError,
+  createFilterExtension,
+} from '@vulcan-sql/core';
+
+import axios from 'axios';
+
+export const RestApiCallerFilter: FunctionalFilter = async ({
+  args,
+  value,
+}) => {
+  if (!args['url']) throw new InternalError('url is required');
+
+  const url = args['arg'] && args['arg'] === ':id' ? `${args['url']}/${value}` : args['url'];
+  const httpMethod = args['method'] || 'get';
+  let options: any = {
+    url: url,
+    method: httpMethod,
+    params: args['arg'] && args['arg'] !== ':id' ? { [args['arg']]: value } : {},
+  }
+  if (args['body']) {
+    options = {...options, body: JSON.parse(args['body'])}
+  }
+  if (args['headers']) {
+    options = {...options, headers: JSON.parse(args['headers'])}
```

Comment:
After discussing with @cyyeh and @onlyjackfrost, we decide to design to pass the `header`, `query, `body`, and `path` from the `value` through the pipe operator, and `rest_api` only pass the `url` keyword argument in the filter function, looks like below

If the URL need to pass the path parameter e.g: `http://localhost:3000/users/:id/workspaces/:sqlName` , then the SQL will like:

```sql
{% set value = { "path": { "id": 1, "sqlName": "ws_1"} }
SELECT {{ value | reset_api(url=http://localhost:3000/users/:id/workspaces/:sqlName) }}
```

If the URL is `http://localhost:3000/users` and could query by query string:

```sql
{% set value = { "query": { "name": "andy" } }
SELECT {{ value | reset_api(url=http://localhost:3000/users) }}
```

If the URL is `http://localhost:3000/users` and could pass the value by header:

```sql
{% set value = { "header": { "x-canner-token": "xxxxx" } }
SELECT {{ value | reset_api(url=http://localhost:3000/users) }}
```

And also update the sample in the test cases

If the URL is `http://localhost:3000/users` and supports `POST` method and pass by `body`: 

```sql
{% set value = { "body": { name: "andy" } }
SELECT {{ value | reset_api(url=http://localhost:3000/users, method="POST") }}
```